### PR TITLE
Packs zip bundle

### PR DESCRIPTION
**Pull request by liorblob**
## Status
- [ ] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/24656

## Description
Downloading the pack zips from the build bucket in GCP and uploading them to the artifacts.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -581,6 +581,24 @@ jobs:
...
+          command: |
+            if [ $CIRCLE_NODE_INDEX -ne 0 ] ;
+              then
+                echo 'Skipping - running only in container number 0.'
+                exit 0;
+            fi
+            GCS_PATH=$(mktemp)
+            ZIP_FOLDER=$(mktemp -d)
+            echo $GCS_MARKET_KEY > $GCS_PATH
+            python3 ./Tests/Marketplace/zip_packs.py -b 'marketplace-ci-build' -z $ZIP_FOLDER -a $CIRCLE_ARTIFACTS -s $GCS_PATH -n $CIRCLE_BUILD_NUM -br $CIRCLE_BRANCH
```
Comment (IkaDemisto) [inline .circleci/config.yml:595]: What about production bucket `marketplace-dist` that runs on master? Don't we need this zip also in master?

File: Tests/Marketplace/zip_packs.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+from concurrent.futures import ThreadPoolExecutor
+import shutil
+from zipfile import ZipFile
+from Tests.Marketplace.marketplace_services import init_storage_client, GCPConfig, IGNORED_FILES, PACKS_FULL_PATH
+
+from demisto_sdk.commands.common.tools import print_error, print_success, print_warning, LooseVersion
+
+ARTIFACT_NAME = 'zipped_packs.zip'
+MAX_THREADS = 4
+GCP_PATH = 'content/builds'
```
Comment (IkaDemisto) [inline Tests/Marketplace/zip_packs.py:12]: If we want to include production bucket, this should be configurable.

File: Tests/Marketplace/zip_packs.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    circle_build = option.circle_build
+    branch_name = option.branch_name
+
+    # google cloud storage client initialized
+    storage_client = init_storage_client(service_account)
+    storage_bucket = storage_client.bucket(storage_bucket_name)
+
+    if not circle_build or not branch_name:
+        # Ignore build properties
+        circle_build = ''
```
Comment (IkaDemisto) [inline Tests/Marketplace/zip_packs.py:145]: Is it for local development?

File: Tests/Marketplace/zip_packs.py [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+        branch_name: The branch name of the build.
+        circle_build: The number of the circle ci build.
+
+    Returns:
+        zipped_packs: A list of the downloaded packs paths and their corresponding pack names.
+    """
+    zipped_packs = []
+    with ThreadPoolExecutor(max_workers=MAX_THREADS) as executor:
+        for pack in os.scandir(PACKS_FULL_PATH):  # Get all the pack names
+            if pack.name in IGNORED_FILES + GCPConfig.CORE_PACKS_LIST:
```
Comment (IkaDemisto) [inline Tests/Marketplace/zip_packs.py:73]: Are you sure that need to ignore core packs?

File: Tests/Marketplace/zip_packs.py [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+        storage_bucket: The GCP bucket to download from.
+        destination_path: The path to download the packs to.
+        branch_name: The branch name of the build.
+        circle_build: The number of the circle ci build.
+
+    Returns:
+        zipped_packs: A list of the downloaded packs paths and their corresponding pack names.
+    """
+    zipped_packs = []
+    with ThreadPoolExecutor(max_workers=MAX_THREADS) as executor:
```
Comment (IkaDemisto) [inline Tests/Marketplace/zip_packs.py:71]: Nice

File: Tests/Marketplace/zip_packs.py (hunk 3 above)
Comment (liorblob) [reply Tests/Marketplace/zip_packs.py:178]: nope, if it's the master bucket then it should not be there

**Pull request merged**